src/renderer/services/binance/transaction.ts

```diff
@@ -1,61 +1,59 @@
 import * as RD from '@devexperts/remote-data-ts'
 import { Address } from '@thorchain/asgardex-binance'
+import { AssetAmount, Asset } from '@thorchain/asgardex-util'
 import * as A from 'fp-ts/lib/Array'
 import * as O from 'fp-ts/lib/Option'
 import * as FP from 'fp-ts/lib/pipeable'
 import * as Rx from 'rxjs'
-import { catchError, map, startWith, switchMap, tap } from 'rxjs/operators'
+import { catchError, map, startWith, switchMap } from 'rxjs/operators'
 
 import { observableState } from '../../helpers/stateHelper'
 import { ClientState } from './service'
+import { TransferRD } from './types'
+import { getBinanceClient } from './utils'
 
-type Transaction = {
-  code: number
-  hash: string
-  log: string
-  ok: boolean
+const { get$: txRD$, set: setTxRD } = observableState<TransferRD>(RD.initial)
+
+type SendTxParams = {
+  to: Address
+  amount: AssetAmount
+  asset: Asset
+  memo?: string
 }
 
-const { get$: transaction$, set: setTransaction } = observableState<RD.RemoteData<Error, Transaction>>(RD.initial)
+const tx$ = ({
+  clientState$,
+  to,
+  amount,
+  asset: { symbol },
+  memo
+}: { clientState$: ClientState } & SendTxParams): Rx.Observable<TransferRD> =>
+  clientState$.pipe(
+    map(getBinanceClient),
+    switchMap((r) => (O.isSome(r) ? Rx.of(r.value) : Rx.EMPTY)),
+    switchMap((client) =>
+      memo
+        ? Rx.from(client.vaultTx(to, amount.amount().toNumber(), symbol, memo))
+        : Rx.from(client.normalTx(to, amount.amount().toNumber(), symbol))
+    ),
+    map(({ result }) => O.fromNullable(result)),
+    map((transfers) => RD.fromOption(transfers, () => Error('Transaction: empty response'))),
+    map((transfers) =>
+      FP.pipe(
+        transfers,
+        RD.map(A.head),
+        RD.chain((transfer) => RD.fromOption(transfer, () => Error('Transaction: no results received')))
+      )
+    ),
```

Comment:
now with `LiveData` we can abstract of inner `RD` in a stream and reolace this with just 2 lines. just take a look on how much pretty it is :) 
```typescript
liveData.map(A.head),
liveData.chain(liveData.fromOption(() => Error('Transaction: no results received'))),
```